test/RestrictedNORI-transfer.test.ts

```diff
@@ -0,0 +1,531 @@
+        ];
+        const expectedInvestor1ScheduleDetailsAfterTransfer = [
+          {
+            tokenHolder: investor1,
+            scheduleTokenId: projectId1,
+            balance: BigNumber.from(restrictedAmounts[0]),
+          },
+          {
+            tokenHolder: investor1,
+            scheduleTokenId: projectId2,
+            balance: BigNumber.from(restrictedAmounts[1]),
+          },
+        ];
+
+        for (const [
+          index,
+          scheduleDetail,
+        ] of supplierScheduleDetailsAfterTransfer.entries()) {
+          compareScheduleDetailForAddressStructs(
+            scheduleDetail,
+            expectedSupplierScheduleDetailsAfterTransfer[index]
+          );
+        }
+
+        for (const [
+          index,
+          scheduleDetail,
+        ] of investor1ScheduleDetailsAfterTransfer.entries()) {
+          compareScheduleDetailForAddressStructs(
+            scheduleDetail,
+            expectedInvestor1ScheduleDetailsAfterTransfer[index]
+          );
+        }
+        const scheduleSummariesAfterTransfer =
+          await rNori.batchGetScheduleSummaries([projectId1, projectId2]);
+
+        for (const scheduleSummary of scheduleSummariesAfterTransfer) {
+          compareScheduleSummaryStructs(scheduleSummary, {
+            tokenHolders: [investor1],
+          });
+        }
+        const [
+          supplierScheduleSetAfterTransfer,
+          investorScheduleSetAfterTransfer,
+        ] = await Promise.all([
+          rNori.getScheduleIdsForAccount(supplier),
+          rNori.getScheduleIdsForAccount(investor1),
+        ]);
+        expect(supplierScheduleSetAfterTransfer.length).to.equal(0);
+        expect(
+          investorScheduleSetAfterTransfer.map((scheduleId) =>
+            scheduleId.toHexString()
+          )
+        ).to.have.deep.members([
+          BigNumber.from(projectId1).toHexString(),
+          BigNumber.from(projectId2).toHexString(),
+        ]);
+      });
+    });
+  });
+  describe('failure', () => {
+    it('should not allow an unapproved account even with DEFAULT_ADMIN_ROLE or SCHEDULE_CREATOR_ROLE to transfer tokens', async () => {
+      const removalDataToList = [
+        {
+          amount: 100,
+          vintage: 2018,
+        },
+      ];
+      const testSetup = await setupTestLocal({});
+      const { rNori, hre } = testSetup;
+      const { listedRemovalIds, projectId } =
+        await batchMintAndListRemovalsForSale({
+          testSetup,
+          removalDataToList,
+        });
+      const { supplier, investor1 } = hre.namedAccounts;
+      const restrictedAmount = removalDataToList[0].amount;
+      await restrictRemovalProceeds({
+        testSetup,
+        removalIds: listedRemovalIds,
+        removalAmountsToRestrict: [restrictedAmount],
+      });
+      await expect(
+        rNori
+          .connect(hre.namedSigners.admin)
+          .safeTransferFrom(supplier, investor1, projectId, 50, '0x')
+      ).to.be.reverted;
```

Comment:
with which error?

Reply:
`"ERC1155: caller is not owner nor approved"` -- lost in the churn of changing how/whether we handle approvals. fixed